pglookout/pglookout.py

```diff
@@ -113,7 +113,7 @@ def sighup(self, _signal=None, _frame=None):
             _signal,
             _frame,
         )
-        self.config_reload_pending = True
+        self.config_reload_pending.put(True)
```

Comment:
The boolean value was introduced specifically to remove use of queue, which is unsafe. Could maybe rather do something like this in `main_loop`?
```
                self.load_config()
                self.config_reload_pending = False
```
=>
```
                self.config_reload_pending = False
                try:
                    self.load_config()
                except:
                    self.config_reload_pending = True
                    raise
```

Reply:
That's interesting. It should probably do the job.
BTW in which sense queue is not safe here?

Reply:
Python's `Queue` takes a non-re-entrant lock when it is accessed and if `main_loop` just obtained the lock when we get a `SIGHUP` signal, the signal handler will also try obtaining the lock but fails because the thread itself already holds the lock and the thread will be permanently stuck.

Reply:
Sorry, just went back to the previous change description. Makes sense now. Thanks.